src/components/utils/src/timer.cc

```diff
@@ -64,6 +64,8 @@ timer::Timer::~Timer() {
   StopDelegate();
   single_shot_ = true;
 
+  delegate_.reset();
+  thread_->set_delegate(NULL);
```

Comment:
@anosach-luxoft In `TransportManagerImplTest Disconnect` was an invalid memory access delegate destructor attempted to call the `set_delegate` method the already deleted thread, which resulted in crash under Windows. Yes, there is a check if `thread_` is not NULL, however it does not detect deleted objects. I will change the order of calling of `delegate`_ and `thread_` methoc calls